### PR TITLE
Javascript Capstone Project

### LINKED ISSUE #3: [3pt]  - Add tests for items counter - Student A
Make sure that function for the items counter is covered with unit tests using Jest. Make sure that you have tests for all edge cases.

### LINKED ISSUE #5: [3pt]  - Add comments counter - Student B
Even if the API gives you the number of comments you will create a specific function to calculate it.

Make sure that the correct number is displayed on the Comments pop up.

Respect the following rules:

- Each counter should be implemented as a separate module.
- A counter function should look for specific DOM elements (e.g. for the comments counter it should look for comments) and make the counting based on what is actually displayed on the page.
- A counter function should cover all the edge cases you can think about.

### LINKED ISSUE #6: [3pt]  - Add all items counter on the Homepage - Student A
Even if the API gives you the number of items, you will create a specific function to calculate it.

Make sure that the correct number is displayed on the Homepage.

Respect the following rules:

- Each counter should be implemented as a separate module.
- A counter function should look for specific DOM elements (e.g. for the comments counter it should look for comments) and make the counting based on what is actually displayed on the page.
- A counter function should cover all the edge cases you can think about.

### LINKED ISSUE #8: [4pt]  - Create feature: add new comment - Student B
Implement the "add new comment" form on the Comments pop up according to the wireframe:

Screenshot 2021-06-27 at 22 32 36

When the user clicks on the "Comment" button, the data is recorded in the [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) and the screen is updated.

### LINKED ISSUE #9: [3pt]  - Create feature: add  new like - Student A
When the user clicks on the Like button of an item (on Homepage), the interaction is recorded in the Involvement API and the screen is updated.

### LINKED ISSUE #11: [3pt]  - Display comments for a given item on the Comments pop-up - Student B
When the popup loads, the webapp retrieves data from the [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) to show the item comments.

Screenshot 2021-06-27 at 22 21 00

This task **does not** include the counter of the comments.

### LINKED ISSUE #12: [4pt]  - Display number of likes for each item on the Homepage - Student A
When the page loads, the webapp the [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) to show the item likes and combines them with the data from the base API.

Remember that your page should make only 2 requests:
- one to the base API
- and **one** to the Involvement API.

This task **does not** include displaying the likes button (heart icon on the wireframe) for each item.

### LINKED ISSUE #14: [4pt]  -Display Comments pop up with selected item's details - Student B
Create a comments popup only with the top section (displaying details of the selected item)

Screenshot 2021-06-27 at 22 10 45

When the popup loads, the webapp retrieves data from the selected API and shows details about the selected item.

Add event to button prepared by your teammate that will open your popup. You need to communicate with your team member to make sure that you do not block each other (most likely the easiest solution is to merge the list of items feature to the dev branch before you try to add that event).

### PR DESCRIPTION
**Pull request by CarlosIgreda**
image

For this Capstone I:

✅ Used GitFlow
✅ Used linters
✅ Used basic JavaScript syntax
✅ Used JavaScript to manipulate DOM elements
✅ Used JavaScript events
✅ Used objects to store and access data
✅ Documented my work in a professional way
✅ Worked with proper ES6 syntax
✅ Used npm as a software packages system
✅ Used webpack to bundle JavaScript
✅ Used POSTMAN and jest
✅ Sent and received data from an API
✅ Used API documentation
✅ Made JavaScript code asynchronous
✅ Applied styling to the website
✅ Used callbacks and promises

### CODE REVIEW

File: src/modules/countLikes.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const countLikes = (countMovies, moviesLength) => {
+  countMovies.textContent += ` (${moviesLength})`;
+};
+
+export default countLikes;
```
Comment (rloterh) [inline src/modules/countLikes.js]: - [ ]  OPTIONAL
When I continue to click on the like button, it keeps the counter only keeps increasing. However, I think you can improve upon your _likes_ feature this way:
    - Like icon should toggle to indicate _like_ or _unlike_ when a particular user clicks on the like icon.
    - Total count of likes per user should be 1.

Can you try achieving this with the help of local storage or other suitable approaches?

File: src/modules/countLike.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    };
+    const moviesLength = 5;
+
+    // Act
+    countLikes(countMovies, moviesLength);
+
+    // Assert
+    expect(countMovies.textContent).toBe('Number of Movies (5)');
+  });
+});
```
Comment (rloterh) [inline src/modules/countLike.test.js]: - [ ] Nice job for adding a test case for _Like_ counter, however, we intend that you provide test cases for other counters such as the _Home_ counter (_Counter for a number of movies on the home screen_) and the _Comment_ counter as well.
 - [ ]  Please also make sure to create tests for **all** possible edge cases for the counter functions, for instance, when the counter value is 0;

**Comment by rloterh** [comment]
Thanks for sharing consent on permission granted to work individually on this project. I'll proceed to review this project.

Cheers! 😃

File: README.md [hunk 3]
```diff
@@ -1,2 +1,169 @@
...
+- **Understand and use JSON**
+- **Make JavaScript code asynchronous**
+- **Use callbacks and promises**
+
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://carlosigreda.github.io/JavaScript-Capstone/)
```
Comment (rloterh) [inline README.md]: - [x] App is not properly deployed hence throwing a 404 error request. Please ensure your app is properly deployed.

**Live demo not displaying correctly:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Carlos, 

You did extremely great on this project! 👏 👏
There are some issues that you still need to work on to prepare your project for the assessment but you are almost there 
🏁 🚦 ⛪ 🚜 🏃 🌴 🏫 🌳  🏘️ 

## Highlights 👍
- Nice job adding counters!
- Counter values are stored and fetched correctly using Involvement API!
- Awesome UI design, matches wireframe design!

## Suggested changes
_Kindly check my inline comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._ 😃 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 3 above)
Comment (CarlosIgreda) [reply README.md]: Done it @rloterh, Deployment added.

Comment (rloterh) [reply README.md]: > Done it @rloterh, Deployment added.
Please resubmit your project for review once changes are completed! 😃

Comment (CarlosIgreda) [reply README.md]: I do have not enough time to finish testing, can I submit all the work that I did please, I was alone in this week without help. I did part that wasn't mine like the comments section, counters, and pop-ups.

File: src/modules/countLike.test.js (hunk 2 above)
Comment (Olamarx) [reply src/modules/countLike.test.js]: > * [ ]  Nice job for adding a test case for _Like_ counter, however, we intend that you provide test cases for other counters such as the _Home_ counter (_Counter for a number of movies on the home screen_) and the _Comment_ counter as well.
> * [ ]   Please also make sure to create tests for **all** possible edge cases of the counter functions, for instance, when the counter value is 0;

- [ ] I see you have removed the whole test in its entirety, kindly know that it is a core requirement that can't be overlooked, From our chat in Slack, I understand that you are quite concerned about the time but take your time to work sufficiently and understand it. You can always write to Students' Success to even help you if you work on this project over the weekend and then pass it on Monday, Students' Success can help you, group to the new week, I have experienced this before. They are there for our progress. I hope you understand. Also, remember to fix the whole recommendation of the previous reviewer.

File: README.md (hunk 3 above)
Comment (Olamarx) [reply README.md]: > * [x]  App is not properly deployed hence throwing a 404 error request. Please ensure your app is properly deployed.
> 
> **Live demo not displaying correctly:** image

Nice fixing this.

Comment (rloterh) [reply README.md]: @CarlosIgreda 
For the deadline being so close, kindly proceed to resubmitting your work.

Cheers! 😃

Comment (CarlosIgreda) [reply README.md]: Done it @rloterh, I resubmitted my work 🤓

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @CarlosIgreda, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Deployment of App ✔️
- Readme file professional ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rloterh** [comment]
@CarlosIgreda 
Can you add the tests and those removed  ASAP?

**Comment by CarlosIgreda** [comment]
@rloterh It's already there buddy.

**Comment by rloterh** [comment]
> @rloterh It's already there buddy.

Checking

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Carlos,  

In my opinion, your project is ready for the presentation! Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲  Counter values are stored and fetched correctly using Involvement API!
🎲  Nice job adding counters!
🎲  Comment modal properly implemented!

Cheers..Keep rocking! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by CarlosIgreda** [comment]
Thank you so much, I really did a great effort working on this project alone, but now I can be calmer. I really appreciate your time following this PR and your feedback. Have a nice weekend 🚀🤓 .

**Comment by ahmetbozaci** [comment]
### Hello @CarlosIgreda 
### _This is an additional review as a result of the second opinion request._ 

- It looks like your project is approved. If you have any questions, please don't hesitate to contact me.


___



### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_
